src/ui/public/agg_types/metrics/derivative.js

```diff
@@ -0,0 +1,142 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import orderAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import $ from 'jquery';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+
+export default function AggTypeMetricDerivativeProvider(Private) {
+  const DerivativeAggType = Private(AggTypesMetricsMetricAggTypeProvider);
+  const AggConfig = Private(VisAggConfigProvider);
+  const Schemas = Private(VisSchemasProvider);
+
+  const aggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+  const orderAggSchema = (new Schemas([
+    {
+      group: 'none',
+      name: 'orderAgg',
+      title: 'Order Agg',
+      aggFilter: aggFilter
+    }
+  ])).all[0];
+
+  return new DerivativeAggType({
+    name: 'derivative',
+    title: 'Derivative',
+    makeLabel: function (aggConfig) {
```

Comment:
you mean makeLabel ? I would argue that it would be less readable after. I usually look at the following: 
- can the code be reused by something else ? if yes, separate out in a new file
- is the original file too complex or long ? if yes, separate out
- is it impossible to test ? if yes, separate out

but i think on all above the answer is no (i can test, the file is not really long, and this seems specific to just this agg, wont be reused elsewhere)